src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/MobDropListener.java

```diff
@@ -31,11 +33,16 @@ public void onEntityKill(EntityDeathEvent e) {
         if (e.getEntity().getKiller() != null) {
             Player p = e.getEntity().getKiller();
             ItemStack item = p.getInventory().getItemInMainHand();
-
+            int random = ThreadLocalRandom.current().nextInt(100);
+            
             Set<ItemStack> customDrops = SlimefunPlugin.getRegistry().getMobDrops(e.getEntityType());
-            if (customDrops != null && !customDrops.isEmpty()) {
-                addDrops(p, customDrops, e.getDrops());
-            }
+            if (customDrops != null && !customDrops.isEmpty()) 
+            	for(ItemStack is : customDrops) {
+            		SlimefunItem sfi = SlimefunItem.getByItem(is);
+            		if (sfi instanceof ChanceDrop && ((ChanceDrop)sfi).getChance() >= random) 
+            				addDrops(p, customDrops, e.getDrops());
```

Comment:
Also, since this is in a for loop... wouldn't this add the entire drops list multiple times instead of the items one by one?

Reply:
I see what you're saying here, however; in testing I did not ever receive any extra drops no matter how many other items the creature in question dropped.   It may have become an issue if the creature dropped multiple slimefun items...   Moved the logic to the addDrops() method to just skip them if they fail the chance check.

Reply:
Yes, you are looping through "customDrops", so this issue would only appear when multiple Slimefun items were at play here.